src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotator.java

```diff
@@ -209,21 +225,52 @@ public void apply(final VariantContext vc, final ReadsContext readsContext, fina
 
         // if the reference is present and base is not ambiguous, we can annotate
         if (refContext.getBases().length ==0 || BaseUtils.simpleBaseToBaseIndex(refContext.getBase()) != -1 ) {
-            //TODO remove this filter and update the tests, this implementation filters out reads that start in a spanning deleting according to variant context in order to match gatk3,
-            //TODO this will cause the reads to be assigned and annotated in a different manner than the haplotype caller.
-            final List<GATKRead> reads = Utils.stream(readsContext).filter(r -> r.getStart() <= vc.getStart()).collect(Collectors.toList());
-
-            AlleleLikelihoods<GATKRead, Allele> likelihoods = new UnfilledReadsLikelihoods<>( variantSamples, new IndexedAlleleList<>(vc.getAlleles()),
-                    AssemblyBasedCallerUtils.splitReadsBySample(variantSamples, getHeaderForReads(), reads));
-
-            VariantContext annotatedVC = annotatorEngine.annotateContext(vc, fc, refContext, likelihoods, a -> true);
+            final ReferenceContext expandedRefContext = !hasReference() ? refContext :
+                    new ReferenceContext(refContext, new SimpleInterval(vc).expandWithinContig(REFERENCE_PADDING, getBestAvailableSequenceDictionary()));
+            VariantContext annotatedVC = annotatorEngine.annotateContext(vc, fc, expandedRefContext, makeLikelihoods(vc, readsContext), a -> true);
             vcfWriter.add(annotatedVC);
-
         } else {
             vcfWriter.add(vc);
         }
     }
 
+    private AlleleLikelihoods<GATKRead, Allele> makeLikelihoods(final VariantContext vc, final ReadsContext readsContext) {
+        //TODO remove this filter and update the tests, this implementation filters out reads that start in a spanning deleting according to variant context in order to match gatk3,
+        //TODO this will cause the reads to be assigned and annotated in a different manner than the haplotype caller.
+        final List<GATKRead> reads = Utils.stream(readsContext).filter(r -> r.getStart() <= vc.getStart()).collect(Collectors.toList());
+        final AlleleLikelihoods<GATKRead, Allele> result = new AlleleLikelihoods<>(variantSamples,
+                new IndexedAlleleList<>(vc.getAlleles()), AssemblyBasedCallerUtils.splitReadsBySample(variantSamples, getHeaderForReads(), reads));
+
+        final ReadPileup readPileup = GATKProtectedVariantContextUtils.getPileup(vc, readsContext);
+        final Map<String, ReadPileup> pileupsBySample = readPileup.splitBySample(getHeaderForReads(), "__UNKNOWN__");
+
+        final Allele refAllele = vc.getReference();
+        final List<Allele> altAlleles = vc.getAlternateAlleles();
+        final int numAlleles = vc.getNAlleles();
+
+        // manually fill each sample's likelihoods one read at a time, assigning probability 1 to the matching allele, if present
```

Comment:
Well, probability 1 in log10 space -- can you make a note?

Reply:
done